### PR TITLE
Feature/251 improve catering order time

### LINKED ISSUE #251: Improve catering order time
Make it radio button like this:

Tarjoiluaika
(x) varauksen alkamisaika
( ) _____

Empty value means reservation start time (the first option).

### PR DESCRIPTION
**Pull request by YoYuUm**
Closes #251

### CODE REVIEW

File: app/shared/reservation-form/catering/catering-form/ServingTimeField.js [hunk 1]
```diff
@@ -1,54 +1,65 @@
...
+    controlProps: PropTypes.object,
+    label: PropTypes.string,
+    type: PropTypes.string,
   }
 
   constructor(props) {
     super(props);
     this.state = {
-      startAtReservationTime: !props.input.value,
+      enabledCustomTime: !!props.input.value,
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/ServingTimeField.js]: - [ ] I think `Boolean(props.input.value)` would be more readable.

File: app/shared/reservation-form/catering/catering-form/CateringForm.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ export function UnconnectedCateringForm(props) {
           {renderField(
             'servingTime',
             'time',
-            'Tarjoiluaika',
+            'Klo:',
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/CateringForm.js]: - [ ] Change to `Klo`

**Comment by YoYuUm** [comment]
Fixed

File: app/shared/reservation-form/catering/catering-form/ServingTimeField.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    input: PropTypes.object.isRequired,
+    controlProps: PropTypes.object,
+    label: PropTypes.string,
+    type: PropTypes.string,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      enabledCustomTime: !!props.input.value,
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/ServingTimeField.js]: - [ ] I think `Boolean(props.input.value)` would be more readable.

File: app/shared/reservation-form/catering/catering-form/ServingTimeField.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          </div>
+          <div>
+            <input
+              defaultChecked={this.state.enabledCustomTime}
+              type="radio"
+              id="serving-custom-time"
+              name="customTime"
+              onClick={this.enableCustomTime}
+            />
+            <label className="radio-label" htmlFor="serving-custom-time">Klo:</label>
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/ServingTimeField.js]: - [ ] Change `Klo:` to `Klo`

File: app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+      controlProps: {
+        step: 5,
+      },
+      type: 'time',
+    };
+    return shallow(<ServingTimeField {...defaults} {...props} />);
+  }
+
+  describe('serving at reservation time', () => {
+    it('renders an radio input', () => {
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js]: - [ ] `an` -> `a`

File: app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    it('input has disableCustomTime as onClick function', () => {
+      const wrapper = getWrapper();
+      const instance = wrapper.instance();
+      const input = wrapper.find('#serving-reservation-time');
+      expect(input.prop('onClick')).to.deep.equal(instance.disableCustomTime);
+    });
+  });
+
+  describe('serving at custom time', () => {
+    it('renders an radio input', () => {
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js]: - [ ] `an` -> `a`

File: app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+      const formControl = getFormControlWrapper({ input: { foo: 'bar' } });
+      expect(formControl.prop('foo')).to.equal('bar');
+    });
+
+    it('gets props from controlProps', () => {
+      const formControl = getFormControlWrapper({ controlProps: { foo: 'bar' } });
+      expect(formControl.prop('foo')).to.equal('bar');
+    });
+
+    it('gets disabled prop is disabled custom time', () => {
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js]: - [ ] Do you mean `is` -> `if`?

File: app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+      const wrapper = getWrapper();
+      const instance = wrapper.instance();
+      instance.setState({ enabledCustomTime: false });
+      const formControl = wrapper.find('.serving-time-form-control');
+      expect(formControl.prop('disabled')).to.be.true;
+    });
+
+    it('gets endabled prop is endabled custom time', () => {
```
Comment (evehmane) [inline app/shared/reservation-form/catering/catering-form/ServingTimeField.spec.js]: - [ ] `is` -> `if`, `endabled` -> `enabled`

**Comment by evehmane** [comment]
Looks good, please squash and merge!

**Pull request merged**